File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/MatchedAheadFunction.java
package cz.fb.manaus.reactor.profit.progress.function;

import cz.fb.manaus.core.model.SettledBet;
import org.springframework.stereotype.Component;

import java.time.Instant;
import java.util.Optional;

@Component
public class MatchedAheadFunction implements AheadTimeFunction {

    @Override
    public Optional<Instant> getRelatedTime(SettledBet bet) {
        var matched = bet.getMatched();
        if (matched == null) {
            return Optional.empty();
        } else {
            return Optional.ofNullable(matched.toInstant());
        }
    }

}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/MatchedAheadFunction.kt
package cz.fb.manaus.reactor.profit.progress.function

import cz.fb.manaus.core.model.RealizedBet
import org.springframework.stereotype.Component
import java.time.Instant

@Component
object MatchedAheadFunction : AheadTimeFunction {

    override fun getRelatedTime(bet: RealizedBet): Instant? {
        return bet.settledBet.matched
    }

}
